packages/kit/src/runtime/server/cookie.js

```diff
@@ -27,26 +27,6 @@ export function get_cookies(request, url, trailing_slash) {
 	// Emulate browser-behavior: if the cookie is set at '/foo/bar', its path is '/foo'
 	const default_path = normalized_url.split('/').slice(0, -1).join('/') || '/';
 
-	if (__SVELTEKIT_DEV__) {
-		// TODO this could theoretically be wrong if the cookie was set unencoded?
-		const initial_decoded_cookies = parse(header, { decode: decodeURIComponent });
-		// Remove all cookies that no longer exist according to the request
-		for (const name of Object.keys(cookie_paths)) {
-			cookie_paths[name] = new Set(
-				[...cookie_paths[name]].filter(
-					(path) => !path_matches(normalized_url, path) || name in initial_decoded_cookies
-				)
-			);
-		}
```

Comment:
I don't think that matters — the goal here is to get a warning when `cookies.get('x')` results in an undefined after a `cookies.set('x', 'y')` due to `path` being omitted, so that the `cookies.set` code can be fixed. That's true whether or not the cookie was deleted or expired in the meantime. In other words the state of the cookies themselves isn't the thing that matters

Reply:
The code that checks if the warning should be emitted checks if there's anything stored in `cookie_paths` with that cookie name. If there is something, it will warn. That something would be there erroneously if it wouldn't be removed when the current incoming headers from the browser would be taken into account, and no-longer-existing cookies, wouldn't be removed:
1. `cookies.set('x', '/')`
2. cookie expires
3. new request coming in, 'x' is no longer in the cookie headers, but isn't removed
4. `cookies.get('x')` shows a wrong warning that there's an existing cookie at path `/`, which it no longer isn't

Reply:
true — but that indicates we need to tweak the logic so that the warning is only shown if the cookie was set at a more specific path. if the cookie was set at `/` then the warning should never be shown

Reply:
https://github.com/sveltejs/kit/pull/9298/commits/b215553715e4767d0caf49358eb042dedeb24779

Reply:
That change makes sense, although the cases where this whole condition is true will be very rare - most people probably set some cookie and only read it on a subsequent request, which this no longer catches. If we reinstantiate the "delete things" logic we would catch more, and I'm not sure if there's any false positive there, because the removal of cookies worked like "remove every cookie being on the path to the current url". So if you have set your cookie on path `/` previously and the cookie is no longer existing on `/foo`, that logic would remove `/` (but not for example a cookie on path `/bar`).

Reply:
`cookie_paths` persists as long as the dev server is alive, so it'll still be caught

Reply:
🤦

Reply:
The change could still be result in a false positive (although that's a very contrived scenario): If you have set cookie `foo` at path `/foo`, then that cookie expires (but we don't delete it from our lookup map), then try to get `foo` at path `/bar`, then we get a false positive warning that `foo` is available at `/foo`, which it no longer isn't.

Reply:
That's not a false positive though — what matters isn't whether the cookie is 'available', it's whether it was _previously set at a different path_. If I set a cookie at `/foo` and later try and get it at `/bar` then that's almost certainly a bug in my code, completely irrespective of whether the cookie expired in the meantime

Reply:
Yeah I can get behind that - it's more likely that this is a bug than intended.